app/js/directives.js

```diff
@@ -1447,6 +1451,74 @@ angular.module('myApp.directives', ['myApp.filters'])
         });
       });
 
+      if (Config.Navigator.ffos) {
+
+        navigator.getUserMedia = ( navigator.getUserMedia || navigator.webkitGetUserMedia || navigator.mozGetUserMedia || navigator.msGetUserMedia);
+
+        voiceRecord.on('touchstart', function(e) {
+          navigator.getUserMedia({audio : true}, function(stream){
+            var start = Date.now();
+            var touch = null;
+
+            audioStream = stream;
+            audioRecorder = new MediaRecorder(stream);
+
+            var interval = setInterval(function(){
+              var time = (new Date());
+
+              time.setTime(Date.now() - start);
+
+              $scope.$apply(function(){
+                $scope.$parent.$parent.voiceRecorder.time = (time.getMinutes() < 10 ? '0' : '') + time.getMinutes() + ':' + (time.getSeconds() < 10 ? '0' : '') + time.getSeconds();
+              });
+            }, 1000);
+
+            $scope.$apply(function(){
+              $scope.$parent.$parent.voiceRecorder.time = '00:00';
+              $scope.$parent.$parent.voiceRecorder.recording = interval;
+            });
+
+            audioRecorder.start();
+
+            console.log('recording now!');
+
+          }, function(e){
+            console.error(e);
+          });
+        });
+
+        voiceRecord.on('click', function(){
+          if (audioRecorder) {
+            $scope.$parent.$parent.voiceRecorder.processing = true;
+
+            audioRecorder.ondataavailable = function(e){
```

Comment:
On my Flame (FxOS 2.5 nightly from 2015-09-13), sometimes the `dataavailable` event has already fired when we get here, which is the source of the bug (=> the audio is never sent, and the recording ribbon stays active with no way to disable it). `touchend` (which stops the recording) fires before `click` (which sets the `dataavailable` handler), and if processing the blob finishes before `click` fires, `audioRecorder.ondataavailable` is never called.

I can't add commits to the PR, but [this patch](https://gist.github.com/wehlutyk/eff0a904fb25ae38afb6) fixes the problem using a promise (L17 on the diff also prevents a new recording from being started when previous audio is being processed).

Then again `touchend` firing before `click` might be something new. @TitanNano can you see if it's the same on your version? (Because if not my patch will break on L26.)